tutorials/calibration_flow/AWQ.py

```diff
@@ -0,0 +1,145 @@
+
+    if compile:
+        model = torch.compile(model)
+
+    # eval on wikitext2
+    testenc = load_dataset("wikitext", "wikitext-2-raw-v1", split="test")
+    testenc = tokenizer("\n\n".join(testenc["text"]), return_tensors="pt")
+    testenc = testenc.input_ids.to(model.device)
+    nsamples = testenc.numel() // max_length
+    model = model.eval()
+    # calculate perplexity
+    nlls = []
+    for i in tqdm(range(nsamples), desc="evaluating..."):
+        batch = testenc[:, (i * max_length) : ((i + 1) * max_length)].to(
+            model.device
+        )
+        with torch.no_grad():
+            lm_logits = model(batch).logits
+        shift_logits = lm_logits[:, :-1, :].contiguous().float()
+        shift_labels = testenc[
+            :, (i * max_length) : ((i + 1) * max_length)
+        ][:, 1:]
+        loss_fct = torch.nn.CrossEntropyLoss()
+        loss = loss_fct(
+            shift_logits.view(-1, shift_logits.size(-1)), shift_labels.view(-1)
+        )
+        neg_log_likelihood = loss.float() * max_length
+        nlls.append(neg_log_likelihood)
+
+    ppl = torch.exp(torch.stack(nlls).sum() / (nsamples * max_length))
+    
+    return ppl
+
+
+parser = argparse.ArgumentParser(description="Evaluate a model with the specified parameters.")
+    
+
+# Optional arguments with default values
+parser.add_argument("repo", type=str, help="Repository ID of the model.")
+parser.add_argument("quant", type=str, help="Quantization method or file path.",choices=["uint4", "int8"])
+parser.add_argument("--calibration_size", type=int, default=100, help="Calibration size. Default is 100.")
+parser.add_argument("--group_size", type=int, default=128, help="Group size to use for weights. Default is '128'")
+parser.add_argument("--device", type=str, default="cuda", help="Device to run the evaluation on. Default is 'cuda'.")
+parser.add_argument("--precision", type=str, default="bfloat16", help="Precision type. Default is 'bfloat16'.")
+parser.add_argument("--max_length", type=int, default=2048, help="Maximum length for evaluation. Default is 2048.")
+parser.add_argument("--compile", action="store_true", help="Flag to indicate if compilation is required.")
+
+args = parser.parse_args()
+
+# Convert precision argument to torch dtype
+precision_dtype = getattr(torch, args.precision, torch.bfloat16)
+
+awq = wikitext2_ppl(
+    repo_id=args.repo,
+    quant="awq-"+args.quant,
+    calibrate_size=args.calibration_size,
+    group_size= args.group_size,
+    device=args.device,
+    precision=precision_dtype,
+    max_length=args.max_length,
+    compile=args.compile
+)
+
+aqt = wikitext2_ppl(
+    repo_id=args.repo,
+    quant=args.quant,
+    calibrate_size=args.calibration_size,
+    group_size= args.group_size,
+    device=args.device,
+    precision=precision_dtype,
+    max_length=args.max_length,
+    compile=args.compile
+)
+print(f"AWQ Perplexity: {awq.item():.5f}")
+print(f"Affine quantized Perplexity: {aqt.item():.5f}")
```

Comment:
can you add a README.md under prototype/awq, report these numbers and also report the performance numbers with torchao/_models/llama/generate.py

Reply:
Ive been using my own eval script (basically the file in the tutorials AWQ.py). Do we want to integrate it within the actual one

Reply:
what is the main issue with the existing one? is the issue about number of samples is too small?

Reply:
I will retry with the generate script to see if it works now. Previously I thought that might be the case.